modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/odata/filter/expression/field/predicate/provider/FieldPredicateProviderTracker.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.object.rest.internal.odata.filter.expression.field.predicate.provider;
```

Comment:
I'm just curious, why are we putting this implementation in the `object-rest-impl` module instead of `object-service`?

Reply:
If we move irt to the service we will have to couple the service with the REST layer via OData... And also we won't be able to share the code in `BinaryExpressionConverterUtil`. Let's discuss it after the holidays.